skip check environment parameter if has no containers.

nazo opened the pull request:
##### SUMMARY
fixed https://github.com/ansible/ansible/commit/179e2882f7e183d75236d1f098795aafb8395c81#commitcomment-24642170 (thanks @Shaita-KrZ)

skip check environment parameter if has no containers.

Fixes #31180.

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
ecs_taskdefinition

##### ANSIBLE VERSION
```
ansible 2.4.0.0
```

##### ADDITIONAL INFORMATION

nazo committed:
skip check environment parameter if has no containers.
fix 179e2882f7e183d75236d1f098795aafb8395c81

daxlab commented on lib/ansible/modules/cloud/amazon/ecs_taskdefinition.py:
```diff
@@ -241,10 +241,11 @@ def main():
-        if 'environment' in container:
-            for environment in container['environment']:
-                environment['value'] = str(environment['value'])
+    if 'containers' in module.params and isinstance(module.params['containers'], list):
```
isn't this more pythonic
```py

if isinstance(module.params.get('containers'), list):
        for container in module.params['containers']:
            if 'environment' in container:
                for environment in container['environment']:
                    environment['value'] = str(environment['value'])

nazo committed:
optimize code

https://github.com/ansible/ansible/pull/31083/files/745d8ef8787463b68c2cc8d443395937dbf4a258#r142025483

s-hertel commented on lib/ansible/modules/cloud/amazon/ecs_taskdefinition.py:
```diff
@@ -241,10 +241,11 @@ def main():
-            for environment in container['environment']:
-                environment['value'] = str(environment['value'])
+    if isinstance(module.params.get('containers'), list):
+        for container in module.params['containers']:
```
LGTM

To simplify this more, you could remove line 244 and do:
`for container in module.params.get('containers', []):`

You could do that below as well:
`for environment in container.get('environment', []):`

s-hertel committed:
Simplify logic and replace str with to_text for compatibility with python 2 and 3.

s-hertel commented:
Thanks @nazo!
